src/go/rpk/pkg/cli/cmd/cluster/config/export.go

```diff
@@ -0,0 +1,187 @@
+	}
+	sort.Strings(keys)
+
+	for _, name := range keys {
+		meta := schema[name]
+		curValue := config[name]
+		template[name] = curValue
+
+		visibility := meta.Visibility
+
+		if visibility == "deprecated" {
+			continue
+		}
+
+		if meta.Visibility == "tunable" && !all {
+			continue
+		}
+
+		// Preface each property with a descriptive comment
+		var comment_tokens []string = make([]string, 0)
+
+		if meta.Example != "" {
+			comment_tokens = append(comment_tokens, fmt.Sprintf("e.g. '%s'", meta.Example))
+		}
+
+		if meta.NeedsRestart {
+			comment_tokens = append(comment_tokens, "restart required")
+		}
+
+		if meta.Nullable {
+			comment_tokens = append(comment_tokens, "may be nil")
+		}
+
+		comment_details := ""
+		if len(comment_tokens) > 0 {
+			comment_details = fmt.Sprintf(" (%s)", strings.Join(comment_tokens, ", "))
+		}
+		file.Write([]byte(fmt.Sprintf("\n# %s%s\n", meta.Description, comment_details)))
+
+		// Compose a YAML representation of the property: this is
+		// done with simple prints rather than the yaml module, because
+		// in either case we have to carefully format values.
+		if meta.Type == "array" {
+			switch x := curValue.(type) {
+			case nil:
+				file.Write([]byte(fmt.Sprintf("%s: []", name)))
+			case []interface{}:
+				if len(x) > 0 {
+					file.Write([]byte(fmt.Sprintf("%s:\n", name)))
+					for _, v := range x {
+						file.Write([]byte(fmt.Sprintf("    - %v\n", v)))
+					}
+				} else {
+					file.Write([]byte(fmt.Sprintf("%s: []", name)))
+				}
+			default:
+				out.Die("Unexpected property value type: %s: %T", name, curValue)
+			}
+
+		} else {
+			scalarVal := ""
+			switch x := curValue.(type) {
+			case int:
+				scalarVal = strconv.Itoa(x)
+			case float64:
+				// Only format floats with a decimal point if they
+				// don't have an integer value currently
+				if x == float64(int64(x)) {
+					scalarVal = strconv.FormatInt(int64(x), 10)
+				} else {
+					scalarVal = strconv.FormatFloat(x, 'f', -1, 64)
+				}
+			case string:
+				scalarVal = x
+			case bool:
+				scalarVal = strconv.FormatBool(x)
+			case nil:
+				// Special case this to avoid a trailing space
+				file.Write([]byte(fmt.Sprintf("%s:\n", name)))
+				continue
+			default:
+				out.Die("Unexpected property value type: %s: %T", name, curValue)
+				continue
+			}
+
+			file.Write([]byte(fmt.Sprintf("%s: %s\n", name, scalarVal)))
+		}
+	}
+
+	return nil
+}
+
+func newExportCommand(fs afero.Fs, all *bool) *cobra.Command {
+	var filename string
+
+	cmd := &cobra.Command{
+		Use:   "export",
+		Short: "Export cluster configuration.",
+		Long: `Export cluster-wide configuration properties
+
+Writes out a YAML representation of the cluster configuration to a file,
+suitable for editing and later applying with the corresponding 'import'
+command.
+
+By default, low level tunables are excluded: use the '--all' flag
+to include all properties including these low level tunables.
+`,
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
nits:
`_ []string` rather than `args` (one linter I will eventually introduce lints against unused function args).

Also probably decent to remove the blank line under this.